src/AtelierTomato.Markov.Storage.Sqlite/SqliteSentenceAccess.cs

```diff
@@ -15,22 +15,22 @@ public SqliteSentenceAccess(IOptions<SqliteAccessOptions> options)
 
 		public async Task DeleteSentenceRange(SentenceFilter filter, string? searchString = null)
 		{
-			if (filter.OIDs.ToList() is null or [] && filter.Authors.ToList() is null or [] && searchString is null)
+			if (filter.OIDs.Count is 0 && filter.Authors.Count is 0 && string.IsNullOrWhiteSpace(searchString))

```

Comment:
what about doing it like this (after changing the Properties to be IReadOnlyList) - this gives the empty filter an easy reading:
`if (filter is { Authors: [], OIDs: [] } && string.IsNullOrWhiteSpace(searchString))`
alternatively, maybe a SentenceFilter.IsEmpty that checks that in SentenceFilter if it's a common idiom? And if it is common also maybe a shared instance of the Empty filter (which should be fine since it's get-only and immutable) to construct from a static property?